target/linux/ath79/dts/qca9531_glinet_gl-x300b.dtsi

```diff
@@ -0,0 +1,172 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "qca953x.dtsi"
+
+/ {
+	compatible = "glinet,gl-x300b", "qca,qca9531";
+	model = "GL.iNet GL-X300B";
+
+	keys {
+		compatible = "gpio-keys-polled";
+
+		poll-interval = <20>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&jtag_disable_pins>;
+
+		button0 {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 3 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		wlan2g {
+			label = "gl-x300b:green:wlan2g";
+			gpios = <&gpio 4 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+
+		wan {
+			label = "gl-x300b:green:wan";
+			gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
+		};
+
+		lte {
+			label = "gl-x300b:green:lte";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+		};
+	};
+    gpio-export {
+		compatible = "gpio-export";
+
+		gpio_lte_power {
+			gpio-export,name = "gpio0";
+			gpio-export,output = <0>;
+			gpios = <&gpio 0 GPIO_ACTIVE_HIGH>;
+		};
+
+		gpio_rs485tx_en {
+			gpio-export,name = "gpio1";
+			gpio-export,output = <0>;
+			gpios = <&gpio 1 GPIO_ACTIVE_HIGH>;
+		};
+
+		gpio_ble_rst {
+			gpio-export,name = "gpio16";
+			gpio-export,output = <0>;
+			gpios = <&gpio 16 GPIO_ACTIVE_HIGH>;
+		};
+
+    };
+    watchdog {
+        compatible = "hw_wdt";
+        dog_en_gpio= <12>;
+        feed_dog_gpio=<2>;
+        feed_dog_interval=<100000000>;
+    };
+
+};
+
+
+&uart {
+	status = "okay";
+    rs485_pin=<1>;
+};
+
+&usb0 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "okay";
+
+	hub_port: port@1 {
+		reg = <1>;
+		#trigger-source-cells = <0>;
+	};
+};
+
+&usb_phy {
+	status = "okay";
+};
+
+&spi {
+	status = "okay";
+	num-cs = <2>;
+	cs-gpios = <0>, <0>;
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+
+		nor_partitions: partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x000000 0x040000>;
+				read-only;
+			};
+
+			partition@40000 {
+				label = "u-boot-env";
+				reg = <0x040000 0x010000>;
+			};
+
+			art: partition@50000 {
+				label = "art";
+				reg = <0x050000 0x010000>;
+				read-only;
+			};
+		};
```

Comment:
Add an inline comment about missing partition that will be found in DTS files.

Reply:
Just closing this without doing anything is not enough to make me forget. But it will influence my mood ...

Please put `/* Firmware partition is found in DTS */` or similar after the last node.

Reply:
I did add a comment, as you saw while continuing your review. I'm working to address your [questions](https://github.com/openwrt/openwrt/pull/4246#discussion_r654963051) about it and will answer you there.

Reply:
Another time, marked as resolved, no change. Since you won't comply to my remarks, I will stop reviewing this now.

Reply:
I'm sorry, I don't understand what you are talking about, I removed the nand partition after establishing that it wasn't present on the device, the details are in the next section of the commit. You complained in your last review about the missing comment you had requested, which was in fact present in the last commit.

I've used OpenWrt for years and dealt with many different members of the core team on a variety of occasions, they have always been welcoming and supportive. Your attitude could at best be described as harsh, and more accurately as vindictive and mean spirited.

The support for this device is quite important to me and I am working very hard to meet the appropriate standards for inclusion.

If you, @adschm, are unwilling to continue your review (for issues actually solved and addressed) then I hope another member of the community will step forwards in your place.

Reply:
I said "Please put /* Firmware partition is found in DTS */ or similar after the last node.".

Did you do that? No.

Reply:
And regarding my tone: Yes, I'm getting harsh at some point. On the other hand, I'm doing tons of device support review. This is really time-consuming. So, what I really hate is having to do the same thing multiple times, because that's just waste of life-time. So, if I've been asking to do a simple thing and you just haven't done it _twice_, I get annoyed.

Reply:
I'm confused and assure you that I am trying to do my best to answer your brief comments. They are no doubt more easily understood by someone much more experienced in this space than myself. I believed that this firmware partition comment request was related to the nand flash that was initially included in this commit and is now removed. I now think I understand what you are requesting and should be able to complete the required edit. It's because, exactly as you said, the firmware partition is in the dts and not the dtsi.

The challenge is that your experience allows you to rapidly review these commits and identify issues, however people that are submitting them, like myself, don't have the same knowledge and consequently don't necessarily understand your feedback.

I will correct this issue tomorrow and hope that you (or someone else) will continue you review. I will point out that if support for this device wasn't very important to me I would have abandoned this commit several revisions ago, to the detriment of anyone who wants to use it. I can understand that you are getting burned out on reviewing these device support commits but there has to be better answer than tearing apart the efforts of people trying to improve OpenWrt.

Reply:
I still don't understand what's so hard to grasp in "Please put /* Firmware partition is found in DTS */ or similar after the last node." ...

Anyway, what I want is this: vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv

```
		nor_partitions: partitions {
			compatible = "fixed-partitions";
			#address-cells = <1>;
			#size-cells = <1>;

			partition@0 {
				label = "u-boot";
				reg = <0x000000 0x040000>;
				read-only;
			};

			partition@40000 {
				label = "u-boot-env";
				reg = <0x040000 0x010000>;
			};

			art: partition@50000 {
				label = "art";
				reg = <0x050000 0x010000>;
				read-only;
			};

			/* Firmware partition is found in DTS */
		};
```

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Reply:
The required edits are very clear to me now, I didn't think you would be reviewing today, I'll get them out to you in the next 2 hours.